packages/maskbook/src/plugins/RedPacket/SNSAdaptor/RedPacketForm.tsx

```diff
@@ -127,7 +125,11 @@ export function RedPacketForm(props: RedPacketFormProps) {
     )
 
     // amount
-    const [rawAmount, setRawAmount] = useState('')
+    const [rawAmount, setRawAmount] = useState(
+        origin?.isRandom
+            ? formatBalance(new BigNumber(origin?.total || 0), origin.token?.decimals ?? 0)
+            : formatBalance(new BigNumber(origin?.total ?? '0').div(origin?.shares || 1), origin?.token?.decimals ?? 0),
```

Comment:
```suggestion
            : formatBalance(new BigNumber(origin?.total ?? '0').div(origin?.shares ?? 1), origin?.token?.decimals ?? 0),
```

Reply:
`origin?.total` might be an empty string, then `||` makes sense.